### PR TITLE
Fix Select's infinitescroll accessibility issues

### PR DESCRIPTION
**Pull request by jeferson-sb**
#### What does this PR do?

Add [aria state attributes](https://www.w3.org/TR/wai-aria-1.1/#aria-setsize) to Select component to better identify active options, the position of each item, and the total amount of options available regardless of how many items are on the DOM (for instance, when using infinite scroll)

#### Where should the reviewer start?
`src/js/components/Select/Select.js`

#### What testing has been done on this PR?
`yarn test`

#### How should this be manually tested?
`yarn storybook`, choose a select story and turn on your screen reader

#### What are the relevant issues?
Related to #5898 

#### Do the grommet docs need to be updated?
No

#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

**Comment by jcfilben** [comment]
Hi @jeferson-sb, thanks for filing this pr! I just wanted to let you know we are currently looking into some other issues (https://github.com/grommet/grommet/issues/5918, https://github.com/grommet/grommet/issues/5917) with Select's accessibility that are affecting the roles and aria labels we have on Select so we are waiting to review/merge this until we know the resolution to the other issues. No additional action is needed on your end at this point, just wanted to let you know why it was taking longer for us to review this.

File: src/js/components/Select/Select.js [hunk 1]
```diff
@@ -291,6 +291,8 @@ const Select = forwardRef(
         <StyledSelectDropButton
           ref={ref}
           a11yTitle={ariaLabel || a11yTitle}
+          aria-expanded={Boolean(open)}
```
Comment (jcfilben) [inline src/js/components/Select/Select.js:294]:
```suggestion
          aria-expanded={open}
```

File: src/js/components/Select/Select.js [hunk 2]
```diff
@@ -291,6 +291,8 @@ const Select = forwardRef(
         <StyledSelectDropButton
           ref={ref}
           a11yTitle={ariaLabel || a11yTitle}
+          aria-expanded={open}
```
Comment (jcfilben) [inline src/js/components/Select/Select.js:294]:
```suggestion
          aria-expanded={Boolean(open)}
```

**Review by jcfilben — APPROVED** [review]
Looks good!

Note to other reviewers: this PR aligns with the direction Amanda and I are going with the Select accessibility work

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**